GetStream/stream-chat-android
# Task/update deprecations deleted messages

### 🎯 Goal

We are removing `DeletedMessageListItemPredicate`, however, due to the footnote decorator there was some extra work to be done.

### 🛠 Implementation details

remove `DeletedMessageListItemPredicate` and pass the new `DeletedMessageVisibility` to the footnote decorator

### 🧪 Testing

1. Apply the patch below
2. Enter a channel
3. Delete a message
4. Check that the message is not visible for all parties

You can also test it using other visibility modes.

<details>
<summary>Patch that changes deleted message visibility before the view is initialized</summary>

```
Index: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt
--- a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt	(revision da867836e959956ce677c760a20410a1223a65e6)
+++ b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt	(date 1655890576500)
@@ -156,7 +156,7 @@
         val calendar = Calendar.getInstance()
         messageListViewModel.apply {
             messageListViewModel.setDeletedMessageVisibility(
-                deletedMessageVisibility = DeletedMessageVisibility.VISIBLE_FOR_CURRENT_USER
+                deletedMessageVisibility = DeletedMessageVisibility.ALWAYS_HIDDEN
             )
             bindView(binding.messageListView, viewLifecycleOwner)
             setDateSeparatorHandler { previousMessage, message ->
```

</details>

The patch below cycles all of the different visibilities automatically **after** the view was initialized. It sets a different deleted message visibility every 5 seconds.

1. Apply the patch below
2. Enter a channel
3. Wait for ~5 seconds
4. Should throw with an error saying: `Adapter was already initialized, please set DeletedMessageVisibility first. If you are using MessageListViewModel, please set the visibility before binding it to MessageListView.`

Further explanation of the issue is in the comments.

<details>
<summary>Patch that cycles deleted messages visibility every 5 seconds after the adapter was initialized</summary>

```
Index: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt
--- a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt	(revision da867836e959956ce677c760a20410a1223a65e6)
+++ b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt	(date 1655890711987)
@@ -45,6 +45,10 @@
 import io.getstream.chat.ui.sample.databinding.FragmentChatBinding
 import io.getstream.chat.ui.sample.feature.common.ConfirmationDialogFragment
 import io.getstream.chat.ui.sample.util.extensions.useAdjustResize
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
 import java.util.Calendar
 
 class ChatFragment : Fragment() {
@@ -159,6 +163,22 @@
                 deletedMessageVisibility = DeletedMessageVisibility.VISIBLE_FOR_CURRENT_USER
             )
             bindView(binding.messageListView, viewLifecycleOwner)
+            CoroutineScope(Dispatchers.Main).launch {
+                repeat(10) {
+                    delay(5000)
+                    messageListViewModel.setDeletedMessageVisibility(
+                        deletedMessageVisibility = DeletedMessageVisibility.ALWAYS_HIDDEN
+                    )
+                    delay(5000)
+                    messageListViewModel.setDeletedMessageVisibility(
+                        deletedMessageVisibility = DeletedMessageVisibility.ALWAYS_VISIBLE
+                    )
+                    delay(5000)
+                    messageListViewModel.setDeletedMessageVisibility(
+                        deletedMessageVisibility = DeletedMessageVisibility.VISIBLE_FOR_CURRENT_USER
+                    )
+                }
+            }
             setDateSeparatorHandler { previousMessage, message ->
                 if (previousMessage == null) {
                     true
```

</details>

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_
